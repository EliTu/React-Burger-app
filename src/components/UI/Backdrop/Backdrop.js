import React from 'react';
import styles from './Backdrop.module.css';
import PropTypes from 'prop-types';

const Backdrop = ({ show, removeBackdrop }) => {
	// CSS Modules styles:
	const { Backdrop } = styles;

	return show && <div className={Backdrop} onClick={removeBackdrop} />;
};

Backdrop.propTypes = {
	show: PropTypes.bool,
	removeBackdrop: PropTypes.func,
};

export default Backdrop;
