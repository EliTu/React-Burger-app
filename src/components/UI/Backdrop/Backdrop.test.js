import React from 'react';
import Backdrop from './Backdrop';
import { shallow } from 'enzyme';

describe('<Backdrop>', () => {
	let onClick = jest.fn();
	let component = shallow(<Backdrop show={true} removeBackdrop={onClick} />);

	it('should render the component without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
	});

	it('should call the click event callback function upon clicking on the button', () => {
		component.simulate('click');

		expect(onClick).toHaveBeenCalled();
	});
});
