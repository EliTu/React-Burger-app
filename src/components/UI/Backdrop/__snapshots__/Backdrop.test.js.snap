// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop> should render the component without errors 1`] = `
<div
  className="Backdrop"
  onClick={[MockFunction]}
/>
`;
